cmd/license-update.go

```diff
@@ -0,0 +1,137 @@
+// Copyright (c) 2015-2022 MinIO, Inc.
+//
+// This file is part of MinIO Object Storage stack
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package cmd
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/fatih/color"
+	"github.com/minio/cli"
+	json "github.com/minio/colorjson"
+	"github.com/minio/mc/pkg/probe"
+	"github.com/minio/pkg/console"
+)
+
+var licenseUpdateCmd = cli.Command{
+	Name:         "update",
+	Usage:        "update the license",
+	OnUsageError: onUsageError,
+	Action:       mainLicenseUpdate,
+	Before:       setGlobalsFromContext,
+	Flags:        append(globalFlags, subnetCommonFlags...),
+	CustomHelpTemplate: `NAME:
+  {{.HelpName}} - {{.Usage}}
+
+USAGE:
+  {{.HelpName}} ALIAS LICENSE-TOKEN
+
+FLAGS:
+  {{range .VisibleFlags}}{{.}}
+  {{end}}
+
+EXAMPLES:
+  1. Update license for cluster with alias 'play'
+     {{.Prompt}} {{.HelpName}} play eyJ0eXAiOiJKV1QiLCJhbGciOiJFUzM4NCJ9.eyJsaWQiOiJjMGRlNDk4OC0wYzA3LTRiYTMtYjY5NS00MDQ2NjllZDQwMWYiLCJzdWIiOiJzaGlyZWVzaCtjMSthZG1pbkBtaW5pby5pbyIsImNhcCI6MjAwMDAsIm9yZyI6IlNoaXJlZXNoLUMxIiwiaXNzIjoic3VibmV0QG1pbi5pbyIsImFpZCI6MSwiaWF0IjoxLjY1NzA4NDEwNDAzNjQxMzQ3MmU5LCJwbGFuIjoiRU5URVJQUklTRSIsImV4cCI6MS42ODg2MjAxMDQwMzY0MTM0NzJlOSwiZGlkIjoiY2Q2NTIwZTctYjg2Ny00YWU2LTkyY2EtNDc5MWI2OWEwY2M3In0.Ya9_HSpog8EhPY1Ckcay5J70_Rms1dNnu4xNlKvwy-8fF6lyF2bqsQMvuDOKaCIYj5w4May8l-1SJ5tC2mQ9Z_ycgCVWpwHGx6h2b7EOAtjGiN6yFEWBLedEScUx34u8
```

Comment:
I think pasting an entire JWT won't really be good for the bash prompt. I think we should take the input slightly differently in this.

This also looks like something is wrong here that we are taking in JWT in this manner.

Reply:
It's very easy to get JWTs wrong when taken as input, because of their sheer length and they wrap around terminals. We have to be careful about this visually as this can simply lead to an uptick in support activity.

Reply:
Yes, will change to accept it as input after running the command.